sdk/digitaltwins/Azure.DigitalTwins.Core/src/DigitalTwinsClient.cs

```diff
@@ -152,13 +152,25 @@ protected DigitalTwinsClient()
         /// </example>
         public virtual async Task<Response<T>> GetDigitalTwinAsync<T>(string digitalTwinId, CancellationToken cancellationToken = default)
         {
-            // Get the digital twin as a stream object
-            Response<Stream> digitalTwinStream = await _dtRestClient.GetByIdAsync(digitalTwinId, null, cancellationToken).ConfigureAwait(false);
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(DigitalTwinsClient)}.{nameof(GetDigitalTwin)}");
+            scope.AddAttribute(nameof(digitalTwinId), digitalTwinId);
+            scope.Start();
 
-            // Deserialize the stream into the specified type
-            T deserializedDigitalTwin = (T)await _objectSerializer.DeserializeAsync(digitalTwinStream, typeof(T), cancellationToken).ConfigureAwait(false);
+            try
+            {
+                // Get the digital twin as a stream object
+                Response<Stream> digitalTwinStream = await _dtRestClient.GetByIdAsync(digitalTwinId, null, cancellationToken).ConfigureAwait(false);
+
+                // Deserialize the stream into the specified type
+                T deserializedDigitalTwin = (T)await _objectSerializer.DeserializeAsync(digitalTwinStream, typeof(T), cancellationToken).ConfigureAwait(false);
 
-            return Response.FromValue<T>(deserializedDigitalTwin, digitalTwinStream.GetRawResponse());
+                return Response.FromValue<T>(deserializedDigitalTwin, digitalTwinStream.GetRawResponse());
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
```

Comment:
is there a `scope.Stop()` for the happy path?

Reply:
There is no Stop API, but during disposal, the activity is implicitly stopped.